shunting-yard.h

```diff
@@ -188,18 +188,43 @@ struct rpnBuilder {
   // * * * * * Static parsing helpers: * * * * * //
 
   // Check if a character is the first character of a variable:
-  static inline bool isvarchar(const char c) {
-    return isalpha(c) || c == '_';
+  // Returns the byte-length of the character
+  // (rest is needed for UTF8 characters)
+  static inline unsigned char isvarchar(const char c, const char** rest) {
+    unsigned char utf8charsize = isUTF8char(c, rest);
+    if(utf8charsize == 0) return isalpha(c) || c == '_' ? true : false;
+    return utf8charsize;
+  }
+
+  // Checks if this is the start of a multi-character unicode character
+  // and if it is, will return the byte-size of the character.
+  // returns zero if it is not a multi-character unicode character
+  // throws a domain_error exception if the character is malformed
+  static inline unsigned char isUTF8char(const char c, const char** rest) {
+    unsigned char counter = 0;
+    if(c & 0x80) { // This is the start of a multi-char unicode character
+      counter++;
+      if(c & 0x40) counter++;
+      if(c & 0x20) counter++;
+      if(c & 0x10) counter++;
+      for(int i = 0; i < counter; i++) {
+        if(!((*rest)[i] & 0x80))
+          throw std::domain_error("Subsequent bytes of unicode character have to be of the form \\b10xxxxxx");
+      }
+    }
+    return counter;
   }
```

Comment:
Here again we could make those changes on the signature:

```suggestion
  static inline uint8_t isUTF8char(const char* str) {
    unsigned char counter = 0;
    char c = *str;
    if(c & 0x80) { // This is the start of a multi-char unicode character
      counter++;
      if(c & 0x40) counter++;
      if(c & 0x20) counter++;
      if(c & 0x10) counter++;
      for(int i = 0; i < counter; i++) {
        if(!(str[i] & 0x80))
          throw std::domain_error("Subsequent bytes of unicode character have to be of the form \\b10xxxxxx");
      }
    }
    return counter;
  }
```